src/main/common/maths.c

```diff
@@ -21,6 +21,44 @@
 #include "axis.h"
 #include "maths.h"
 
+// http://lolengine.net/blog/2011/12/21/better-function-approximations
+// Chebyshev http://stackoverflow.com/questions/345085/how-do-trigonometric-functions-work/345117#345117
+// Thanks for ledvinap for making such accuracy possible! See: https://github.com/cleanflight/cleanflight/issues/940#issuecomment-110323384
+// https://github.com/Crashpilot1000/HarakiriWebstore1/blob/master/src/mw.c#L1235
+#if defined(FAST_TRIGONOMETRY) || defined(EVEN_FASTER_TRIGONOMETRY)
+#if defined(EVEN_FASTER_TRIGONOMETRY)
+#define sinApproxCoef3 -1.666568107e-1f
+#define sinApproxCoef5  8.312366210e-3f
+#define sinApproxCoef7 -1.849218155e-4f
+#else
+#define sinApproxCoef3 -1.666665710e-1f                                          // Double: -1.666665709650470145824129400050267289858e-1
+#define sinApproxCoef5  8.333017292e-3f                                          // Double:  8.333017291562218127986291618761571373087e-3
+#define sinApproxCoef7 -1.980661520e-4f                                          // Double: -1.980661520135080504411629636078917643846e-4
+#define sinApproxCoef9  2.600054768e-6f                                          // Double:  2.600054767890361277123254766503271638682e-6
+#endif
+
+float sin_approx(float x)
+{
+    int32_t xint = x;
```

Comment:
Maybe it would be better to use true range reduction
```
int n = floor(x * (float)(1/(M_PI/2)));   
x = x - n *  (M_PI/2) ;
```
Now n%4 is quadrant of input value and x is reduced to 0 ... PI/2. Switch can be used to get output right (but it's harder to use the fact that used polynomial is even function and works for negative x too).

Also accuracy of the subtraction of n * PI/2  can be improved:
```
x = (x - (n * 3.14159252)) - (n * 1.50995799e-7)
```
First value is closest `float` value lesser than pi, second value is pi - [first value]. This does improve accuracy considerably in my quick tests (also note that is N value is calculated slightly wrong, it won't hurt much - polynomial approximation works even outside of optimized interval) . This won't be probably necessary if resulting precision is about half of float ...